desktop-exporter/app/utils/array-to-tree.ts

```diff
@@ -31,21 +35,24 @@ export function arrayToTree(spans: SpanData[]): TreeItem[] {
       };
     }
 
+    let treeItem = lookup[spanID];
+
     // Note A:
     // If the span has been added to the lookup structure as a missing/incomplete parent
     // on a previous pass (see Note B), update it and mark it present, and remove it from the missing set.
-    if (lookup[spanID].status === SpanDataStatus.missing) {
-      let children = lookup[spanID].children;
-      lookup[spanID] = {
+    if (treeItem.status === SpanDataStatus.missing) {
+      let children = treeItem.children;
+      // Re-assign treeItem as a TreeItem type so that after this if statement,
+      // the type system knows that treeItem can only be a TreeItem
+      treeItem = {
         status: SpanDataStatus.present,
         spanData: spanData,
         children: children,
       };
+      lookup[spanID] = treeItem
```

Comment:
Would it be possible to please add a comment here, saying that you are overwriting the tree item after you set it? There's just a lot going on in this function, in a weird order, and I will 100% forget about it the next time I look at it.

Reply:
Done!